### PR TITLE
Add spec for Bet model

### PR DESCRIPTION
**Pull request by ghost**
Tests winner_total, options.winners and resolve methods for Bet model.

### CODE REVIEW

**Comment by qr8r** [comment]
@YarekTenko I will fix the rubo cop for this one

File: spec/models/bet_spec.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                   amount_bet: 300)
+  end
+
+  let(:ub) do
+    ub1
+    ub2
+    ub3
+  end
+
+  context_string = 'Bet: Superbowl - Atlanta vs New England;\n' \
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: Ive changed the rubocop so that specs wont care if your lines are long. 

In specs its more important to write them so that they are readable and helpful, so writing long context statements is allows. 

This is one of the few cases where saving this string to a variable is bad because it hides the content which should be very obvious for future devs reading this file.

File: spec/models/bet_spec.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    'Betters on Atlanta: Bob ($100), Peter ($50);\n' \
+    'Betters on New England: Jack ($300).'
+
+  context context_string do
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: This context doesnt add a whole lot of value.  generally a `context` statement is used to change the test setup for specific edge cases:

```ruby
describe '#foo' do
  context 'on tuesdays this returns true' do
    let(:date) { :tuesday } 
    it { is_expected.to be truthy }
  end 
  context 'on mondays this returns false' do
    let(:date) { :monday }
    it { is_expected.to be truthy }
  end
end
```
This context is acting more as a description than changing the setup behaviour. If you feel your test needs to be more clear about what it doing, try and write your `let` statements so they are more descriptive about what they return

File: spec/models/bet_spec.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  end
+
+  let(:ub2) do
+    UserBet.create(user:       peter,
+                   bet:        bet1,
+                   bet_option: option1,
+                   amount_bet: 50)
+  end
+
+  let(:ub3) do
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: This is a bad let statement ub3 is not very descriptive

File: spec/models/bet_spec.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  context_string = 'Bet: Superbowl - Atlanta vs New England;\n' \
+    'Betters on Atlanta: Bob ($100), Peter ($50);\n' \
+    'Betters on New England: Jack ($300).'
+
+  context context_string do
+    describe('#winner_total') do
+      subject { bet1.winner_total }
+      context 'Atlanta wins' do
+        it 'returns the total amount of money bet on Atlanta winning (150)' do
+          ub
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: Look up the difference between `let` and `let!`

File: spec/models/bet_spec.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+          option1.update(winner: true)
+          expect(subject).to eq 150
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: This is what we call a "magic number" we try and avoid having numbers floating around in code without any context, because its very easy to become confused about what they represent.

This should be encapsulated in a `let` statement ..... something like `let(:expected_debt_amount) { ... }

File: spec/models/bet_spec.rb [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        end
+      end
+    end
+
+    describe('#options.winners') do
+      subject { bet1.options.winners }
+      context 'Atlanta wins' do
+        it 'returns "Atlanta Wins" option as the winning option' do
+          ub
+          option1.update(winner: true)
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: Why are you changing the object you're testing from within your test.

File: spec/models/bet_spec.rb [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+    describe('#resolve') do
+      subject { bet.resolve }
+      context 'Atlanta wins' do
+        it 'creates entries in Debt model so that: Jack owes Bob 201; Jack owes Peter 99' do
+          ub
+          option1.update(winner: true)
+          bet1.resolve
+          expect(subject.find_by(debtor: jack, creditor: bob).amount).to eq 201
+          expect(subject.find_by(debtor: jack, creditor: peter).amount).to eq 99
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: As much as possible try and avoid multiple expectations per test. 

sometimes its failures and then you will want to use the `:aggregate_failures` feature of rspec

**Review by qr8r — CHANGES_REQUESTED** [review]

File: spec/models/bet_spec.rb [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                   bet_option: option2,
+                   amount_bet: 300)
+  end
+
+  describe('#winner_total') do
+    subject { bet1.winner_total }
+    context 'Atlanta wins' do
+      let(:expected_atlanta_win_total) { 150 }
+      it 'returns the total amount of money bet on Atlanta winning (150)' do
+        users_place_bets
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: Not needed because you define this in a `let!`

File: spec/models/bet_spec.rb [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                   bet:        bet1,
+                   bet_option: option2,
+                   amount_bet: 300)
+  end
+
+  describe('#winner_total') do
+    subject { bet1.winner_total }
+    context 'Atlanta wins' do
+      let(:expected_atlanta_win_total) { 150 }
+      it 'returns the total amount of money bet on Atlanta winning (150)' do
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: The number should be removed here. It could be changed and then we have to update it multiple places. 

`it 'is the correct money bet amount' ` would be enough information

File: spec/models/bet_spec.rb [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    end
+  end
+
+  describe('#resolve') do
+    subject { bet1.resolve }
+    context 'Always' do
+      it 'returns array of newly created debts' do
+        users_place_bets
+        option1.update(winner: true)
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: This should be part of the setup that is relevant to this context block

File: spec/models/bet_spec.rb [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+        expect(subject).to eq Debt.all
+      end
+    end
+    context 'Atlanta wins' do
+      let(:expected_debt_jack_to_bob) { 201 }
+      let(:actual_debt_jack_to_bob) do
+        Debt.find_by(debtor: jack, creditor: bob).amount
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: You should be able to reference values defined in previous let blocks

**Review by qr8r — CHANGES_REQUESTED** [review]
Ping me tomorrow before you start working and we will go over this together :)

File: app/models/bet.rb [hunk 12]
```diff
@@ -3,7 +3,8 @@ class Bet < ApplicationRecord
   has_many :user_bets
   has_many :options, through: :user_bets, source: :bet_option
 
-  def resolve
+  def resolve(winning_option:)
+    winning_option.update(winner: true)
```
Comment (qr8r) [inline app/models/bet.rb]: Whats the difference between `update` and `update!` ?

File: spec/models/bet_spec.rb [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    let(:option) { option1 }
+    it 'is an array of debts' do
+      expect(debts).to all be_a Debt
+    end
+
+    context 'Atlanta wins' do
+      let(:expected_debt) do
+        Debt.find_by!(debtor: jack, creditor: bob, amount: 201)
+      end
+      let(:option) { option1 }
```
Comment (qr8r) [inline spec/models/bet_spec.rb]: Technically this line is not needed  since it is the same as the let statement defined on line 38

**Review by qr8r — DISMISSED** [review]
This is 99% there. Just a couple super minor comments and then Im happy to approve this guy

**Review by qr8r — APPROVED** [review]
